### PR TITLE
Changes necessary for running MSAL inside broker

### PR DESCRIPTION
**Pull request by oldalton**
* Correct broker response handling that validates calling app
* Return refresh token in MSIDTokenResult
* Return WPJ error to broker

### CODE REVIEW

File: IdentityCore/src/MSIDConstants.m [hunk 1]
```diff
@@ -34,6 +34,9 @@
 NSString *const MSID_BROKER_SYMMETRIC_KEY_TAG   = @"com.microsoft.adBrokerKey";
 NSString *const MSID_BROKER_ADAL_SCHEME         = @"msauth";
 NSString *const MSID_BROKER_MSAL_SCHEME         = @"msauthv2";
+NSString *const MSID_BROKER_APP_BUNDLE_ID        = @"com.microsoft.azureauthenticator";
+NSString *const MSID_BROKER_APP_BUNDLE_ID_DF     = @"com.microsoft.azureauthenticator-df";
```
Comment (unpluggedk) [inline IdentityCore/src/MSIDConstants.m]: Q: out of curiosity what is the df here? dogfood??

File: IdentityCore/src/controllers/MSIDLocalInteractiveController.m [hunk 2]
```diff
@@ -79,7 +79,7 @@ - (void)acquireToken:(MSIDRequestCompletionBlock)completionBlock
 
         if (installBrokerResponse)
         {
-            [self promptBrokerInstallWithResponse:installBrokerResponse completionBlock:completionBlock];
+            [self handleBrokerResponse:installBrokerResponse completion:completionBlock];
```
Comment (unpluggedk) [inline IdentityCore/src/controllers/MSIDLocalInteractiveController.m]: Q: will this response ever be non-install broker? I don't see why this was changed from promptBrokerInstall to handleBrokerResponse

File: IdentityCore/src/controllers/MSIDLocalInteractiveController.m [hunk 3]
```diff
@@ -90,6 +90,30 @@ - (void)acquireToken:(MSIDRequestCompletionBlock)completionBlock
     }];
 }
 
+- (void)handleBrokerResponse:(MSIDWebMSAuthResponse *)response completion:(MSIDRequestCompletionBlock)completionBlock
+{
+    if (![NSString msidIsStringNilOrBlank:response.appInstallLink])
```
Comment (unpluggedk) [inline IdentityCore/src/controllers/MSIDLocalInteractiveController.m]: Related to the above comment, handling broker response and prompting install is two different things, am i misunderstanding here?
Perhaps the method name is wrong then? I just don't see a broker response being "install broker"

File: IdentityCore/src/controllers/MSIDLocalInteractiveController.m (hunk 2 above)
Comment (unpluggedk) [reply IdentityCore/src/controllers/MSIDLocalInteractiveController.m]: Or is this meant to be handleBrokerInstallResponse??

File: IdentityCore/src/controllers/ios/MSIDBrokerInteractiveController.m [hunk 4]
```diff
@@ -208,6 +218,27 @@ + (BOOL)completeAcquireToken:(NSURL *)resultURL
     return completionResult;
 }
 
++ (BOOL)responseIsFromBroker:(NSString *)sourceApplication
```
Comment (unpluggedk) [inline IdentityCore/src/controllers/ios/MSIDBrokerInteractiveController.m]: nit: isResponseFromBroker?

File: IdentityCore/src/controllers/ios/MSIDBrokerInteractiveController.m [hunk 5]
```diff
@@ -187,9 +188,18 @@ - (void)saveToPasteBoard:(NSURL *)url
     [appPasteBoard setURL:url];
 }
 
-+ (BOOL)completeAcquireToken:(NSURL *)resultURL
-       brokerResponseHandler:(MSIDBrokerResponseHandler *)responseHandler
++ (BOOL)completeAcquireToken:(nullable NSURL *)resultURL
```
Comment (unpluggedk) [inline IdentityCore/src/controllers/ios/MSIDBrokerInteractiveController.m]: Q: So, is this something developers will eventually need to call (through MSAL/ADAL) in app delegate?

Comment (unpluggedk) [reply IdentityCore/src/controllers/ios/MSIDBrokerInteractiveController.m]: yeah that is what i meant :p
something the they will call msal that will call this :)

File: IdentityCore/src/controllers/MSIDLocalInteractiveController.m (hunk 2 above)
Comment (oldalton) [reply IdentityCore/src/controllers/MSIDLocalInteractiveController.m]: Changed naming, please take a look.

File: IdentityCore/src/controllers/MSIDLocalInteractiveController.m (hunk 3 above)
Comment (oldalton) [reply IdentityCore/src/controllers/MSIDLocalInteractiveController.m]: Changed naming, please take a look.

File: IdentityCore/src/controllers/ios/MSIDBrokerInteractiveController.m (hunk 4 above)
Comment (oldalton) [reply IdentityCore/src/controllers/ios/MSIDBrokerInteractiveController.m]: Changed.

**Pull request merged**